docs/technical-overview.md

```diff
@@ -0,0 +1,250 @@
+# Podium Request Portal — Technical Documentation
+
+## Project summary
+
+Podium Request Portal is an open source web-based application with which researchers can place requests for samples or data (including images) from Dutch biobanks and registries, and where these requests can be tracked and traced. The focus of this project is on supporting only essential (generic) steps of these request processes, keeping in mind that additional functionality might need to be added later on.
+
+Project source code is available in the following repositories:
+- https://github.com/thehyve/podium-registry — configuration registry and system health monitoring webapp;
+- https://github.com/thehyve/podium — core functionality.
+
+Project codebase is maintained by The Hyve.
+For technical questions you can contact alexey@thehyve.nl or gijs@thehyve.nl
+
+
+## Purpose of the document
+
+This document is intended to be a succinct technical reference that facilitates updating, testing and further development of the Podium project, and is aimed at people who plan to contribute to the project codebase.
+High-level architecture overview, component interactions and major dependencies are included, while implementation details that can be trivially derived from the code (e.g. DTOs) and/or autogenerated (e.g. Swagger API definition, database schema), and user-oriented nuances (e.g. visual elements of the web UIs, installation manual) are omitted.
+
+
+## Principal components
+
+### Overview
+
+
+```plantuml
```

Comment:
This diagram is really helpful, but it is not rendered here on github. How do you render the page? Maybe we can store the puml files separately and include a rendered version here?

Reply:
I normally view them VSCode PlantUML plugin, but it can be rendered as an HTML page with SVG diagrams. It should be possible to link diagram files from markdown docs on Github, I'll check whether it is easy to do.

Reply:
I've extracted the diagrams and made them render via the PlantUML proxy — `http://www.plantuml.com/plantuml/proxy?cache=no&src=<path-to-diagram-description-on-github>` — so we don't even need to store any images.

Caveat: when you update the docs, make sure that `<path-to-diagram-description-on-github>` exists and returns the diagram text before you access the full URL. A bit of inconvenience, but I can live with it.